nginx-controller/nginx/configurator.go

```diff
@@ -74,7 +78,8 @@ func (cnf *Configurator) updateCertificates(ingEx *IngressEx) map[string]string
 
 	return pems
 }
-func (cnf *Configurator) generateNginxCfg(ingEx *IngressEx, pems map[string]string) IngressNginxConfig {
+
+func (cnf *Configurator) generateNginxCfg(ingEx *IngressEx, pems map[string]string) []IngressNginxConfig {
```

Comment:
Since we're moving to a single NGINX config per hostname,  we can drop the `IngressNginxConfig`
```go
type IngressNginxConfig struct {
	Upstreams []Upstream
	Servers   []Server
}
```
and use `Server` instead, provided that we moved the array of upstreams to the Server.